torchtitan/components/ft.py

```diff
@@ -59,7 +58,7 @@ def get_dp_info(self, dp_degree: int, dp_rank: int) -> tuple[int, int]:
 
     def set_all_reduce_hook(self, model_parts: list[torch.nn.Module]) -> None:
         def all_reduce_hook(output):
-            dist.all_reduce(output, group=self.replicate_pg, op=ReduceOp.AVG)
+            self.replicate_pg.allreduce(output, opts=ReduceOp.AVG)
```

Comment:
Can we add a comment discussing why we need to use this call instead of the original one? This is less intuitive.

Reply:
Sure I will add a comment, this is due to these changes (https://github.com/pytorch/pytorch/commit/35c45a4a315425a0940dcc040d8262b6331804dc#diff-61109d1cb2a0bd13fc51d678a82666295289da1ec0a1a694e73d9e8c28f51bdcR2885, so the regular `dist.all_reduce()` is no longer compatible

Reply:
@H-Huang can we fix this in the torchft side? We should just be able to ignore the case when there's no return work object iiuc